core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1702,8 +1702,11 @@ public function generateRecordLabel(array $row, ?string $table = null, bool $pro
 				list($strKey, $strTable) = explode(':', $v, 2);
 				list($strTable, $strField) = explode('.', $strTable, 2);
 
+				$objDcaExtractor = DcaExtractor::getInstance($table);
```

Comment:
```suggestion
				$objDcaExtractor = DcaExtractor::getInstance($strTable);
```

Are you sure that this needs to be `$table` and not `$strTable`?

Reply:
`$table` is correct. `$table` is the current DCA table we need to read the relations from, `$strTable` is the target table of the `foreignKey` field.